crates/miden-objects/src/account/storage/mod.rs

```diff
@@ -51,19 +53,43 @@ impl AccountStorage {
     // CONSTRUCTOR
     // --------------------------------------------------------------------------------------------
 
+    /// TODO(named_slots): Remove this temporary API.
+    ///
     /// Returns a new instance of account storage initialized with the provided items.
     ///
     /// # Errors
     ///
     /// Returns an error if:
     /// - The number of [`StorageSlot`]s exceeds 255.
     pub fn new(slots: Vec<StorageSlot>) -> Result<AccountStorage, AccountError> {
+        let slots = slots
+            .into_iter()
+            .enumerate()
+            .map(|(idx, slot)| NamedStorageSlot::new(SlotName::new_index(idx), slot))
+            .collect();
+
+        Self::new_named(slots)
+    }
+
+    /// TODO(named_slots): Rename to new.
+    pub fn new_named(mut slots: Vec<NamedStorageSlot>) -> Result<AccountStorage, AccountError> {
         let num_slots = slots.len();
 
         if num_slots > Self::MAX_NUM_STORAGE_SLOTS {
             return Err(AccountError::StorageTooManySlots(num_slots as u64));
         }
 
+        let mut names = BTreeMap::new();
+        for slot in &slots {
```

Comment:
nit: do we need a `BTreeMap` here? Seems like we use this only to check the uniqueness of the names. If so, maybe `BTreeSet` would work as well?